src/validation.cpp

```diff
@@ -3001,6 +3003,13 @@ bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationState& sta
     if (block.GetBlockTime() > nAdjustedTime + 2 * 60 * 60)
         return state.Invalid(false, REJECT_INVALID, "time-too-new", "block timestamp too far in the future");
 
+    if (BIP102active(nHeight, fSegwitSeasoned)) {
+        if (block.nVersion & HARDFORK_VERSION_BIT)
+            return true;
```

Comment:
This "return true" will re-enable block.nVersion < 4 for blocks after the HF, because that check is later . Is this really what you want?